analytical_engine/apps/simple_path/all_simple_paths_context.h

```diff
@@ -0,0 +1,204 @@
+    std::vector<vid_t> q;
+    if (native_source) {
+      std::set<vid_t> qvisit;
+      vid_t source_gid = frag.Vertex2Gid(source);
+      qvisit.insert(source_gid);
+      q.push_back(source_gid);
+      int index = find_edge_map_index(source_gid);
+
+      struct timeval t;
+      gettimeofday(&t, 0);
+      counter_time -= static_cast<double>(t.tv_sec) +
+                      static_cast<double>(t.tv_usec) / 1000000;
+      Pint_Result(index, 0, q, qvisit, os);
+      gettimeofday(&t, 0);
+      counter_time += static_cast<double>(t.tv_sec) +
+                      static_cast<double>(t.tv_usec) / 1000000;
+      VLOG(0) << "counter_time : " << counter_time << std::endl;
+    }
+    VLOG(0) << "exec_time : " << exec_time << std::endl;
+  }
+
+  void Pint_Result(int from, int depth, std::vector<vid_t>& q,
+                   std::set<vid_t>& qvisit, std::ostream& os) {
+    auto& frag = this->fragment();
+    if (depth == (this->cutoff - 1)) {
+      // VLOG(0) << "in last: " << std::endl;
+      for (auto t : targets) {
+        if (qvisit.count(t) == 1) {
+          continue;
+        }
+        int to = find_edge_map_index(t);
+        if (std::find(edge_map[from].begin(), edge_map[from].end(), to) !=
+            edge_map[from].end()) {
+          for (auto gid : q) {
+            os << frag.Gid2Oid(gid) << " ";
+          }
+          os << frag.Gid2Oid(t) << " " << std::endl;
+        }
+      }
+      return;
+    }
+
+    for (long unsigned int t = 0; t < edge_map[from].size(); t++) {
+      int to = edge_map[from][t];
+      vid_t gid = index2gid(to);
+      if (qvisit.count(gid) == 1) {
+        continue;
+      }
+      qvisit.insert(gid);
+      if (std::find(targets.begin(), targets.end(), gid) != targets.end()) {
+        for (auto v : q) {
+          os << frag.Gid2Oid(v) << " ";
+        }
+        os << frag.Gid2Oid(gid) << " " << std::endl;
+      }
+      q.push_back(gid);
+      Pint_Result(to, depth + 1, q, qvisit, os);
+      q.pop_back();
+      qvisit.erase(gid);
+    }
+  }
+
+  int find_edge_map_index(vid_t gid) {
+    int fid = (int) (gid >> fid_offset);
+    int lid = (int) (gid & id_mask);
+    int ret = 0;
+    for (int i = 0; i < fid; i++) {
+      ret += frag_vertex_num[i];
+    }
+    return ret + lid;
+  }
+
+  vid_t index2gid(int index) {
+    int i = 0;
+    int sum = 0;
+    int sum_last = 0;
+    while (true) {
+      sum += frag_vertex_num[i];
+      if (sum > index) {
+        int lid = index - sum_last;
+        vid_t gid = (vid_t) ((i << fid_offset) | lid);
+        return gid;
+      }
+      i++;
+      sum_last = sum;
+    }
+  }
+
+  oid_t source_id;
+  std::queue<std::pair<vid_t, int>> curr_level_inner, next_level_inner;
+  std::set<vid_t> visit;
+  std::vector<vid_t> targets;
+  std::vector<int> frag_vertex_num;
+  int cutoff;
+  bool source_flag = false;
+  fid_t soucre_fid;
+  vid_t id_mask;
+  int fid_offset;
+  std::vector<std::vector<int>> edge_map;
```

Comment:
add some comments about  `cur_level_inner`, `next_level_inner`, `frag_vertex_num`, `cutoff` and `edge_map` to explain what these for?